ionc/ion_int.c

```diff
@@ -130,51 +129,52 @@ iERR ion_int_compare(ION_INT *iint1, ION_INT *iint2, int *p_result)
 
     if (!iint1) FAILWITH(IERR_INVALID_ARG);
     if (!iint2) FAILWITH(IERR_INVALID_ARG);
+    if (!p_result) FAILWITH(IERR_INVALID_ARG);
     
     if (iint1 == iint2) {
-        diff = 0;
+        *p_result = 0;
         SUCCEED();
     }
 
     IONCHECK(ion_int_is_null(iint1, &is_null1));
     IONCHECK(ion_int_is_null(iint2, &is_null2));
     if (is_null1 || is_null2) {
-        diff = (is_null1 - is_null2);  // TODO : really?
+        *p_result = (is_null1 - is_null2);  // TODO : really?
```

Comment:
I can only speculate, which probably means it should be removed. The behavior is not incorrect.